valhalla/meili/state.h

```diff
@@ -123,6 +124,14 @@ class StateContainer {
     return static_cast<StateId::Time>(columns_.size());
   }
 
+  // Check to see if we have the minimum number of measurements and edge candidates to perform a map
+  // match. We need at least two measurements with a non-zero number of edge candidates.
+  bool HasMinimumCandidates() {
+    size_t num_nonempty =
+        std::count_if(columns_.begin(), columns_.end(), [](Column& col) { return !col.empty(); });
```

Comment:
can you change this to `find_if` and stop immediately when you find one that isnt empty. there is no need to loop over all of the columns

Reply:
Done